pf4j/src/main/java/ro/fortsoft/pf4j/DefaultPluginManager.java

```diff
@@ -658,6 +658,37 @@ protected PluginClasspath createPluginClasspath() {
     	return new PluginClasspath();
     }
 
+    protected PluginStatusProvider createPluginStatusProvider() {
+        return new PluginStatusProvider() {
+
+            @Override
+            public List<String> getEnabledPlugins() {
+                List<String> enabledPlugins = Collections.emptyList();
+                try {
+                    // create a list with plugin identifiers that should be only accepted by this manager (whitelist from plugins/enabled.txt file)
+                    enabledPlugins = FileUtils.readLines(new File(pluginsDirectory, "enabled.txt"), true);
+                    log.info("Enabled plugins: {}", enabledPlugins);
+                } catch (IOException e) {
+                    log.error(e.getMessage(), e);
+                }
+                return enabledPlugins;
+            }
+
+            @Override
+            public List<String> getDisabledPlugins() {
+                List<String> disabledPlugins = Collections.emptyList();
+                try {
+                    // create a list with plugin identifiers that should not be accepted by this manager (blacklist from plugins/disabled.txt file)
+                    disabledPlugins = FileUtils.readLines(new File(pluginsDirectory, "disabled.txt"), true);
+                    log.info("Disabled plugins: {}", disabledPlugins);
+                } catch (IOException e) {
+                    log.error(e.getMessage(), e);
+                }
+                return disabledPlugins;
+            }
+        };
+    }
+
     protected boolean isPluginDisabled(String pluginId) {
     	if (enabledPlugins.isEmpty()) {
```

Comment:
I think that we can remove `enabledPlugins`, `disabledPlugins` from DefaultPluginManager and in `initialize` in pace of
```java
enabledPlugins = statusLists.getEnabledPlugins();
disabledPlugins = statusLists.getDisabledPlugins();
```

we can put 
```java
pluginStatusProvider = createPluginStatusProvider();
```